src/main/java/prover/Prover.java

```diff
@@ -0,0 +1,195 @@
+package prover;
+
+import static algebra.curves.barreto_naehrig.bn254a.BN254aFields.BN254aFr;
+
+import algebra.curves.AbstractG1;
+import algebra.curves.AbstractG2;
+import algebra.curves.barreto_naehrig.bn254a.BN254aBinaryReader;
+import algebra.curves.barreto_naehrig.bn254a.BN254aBinaryWriter;
+import algebra.curves.barreto_naehrig.bn254a.BN254aG1;
+import algebra.curves.barreto_naehrig.bn254a.BN254aG2;
+import algebra.fields.AbstractFieldElementExpanded;
+import configuration.Configuration;
+import io.AssignmentReader;
+// import profiler.utils.SparkUtils;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import org.apache.commons.cli.*;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function2;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.storage.StorageLevel;
+import scala.Tuple2;
+import zk_proof_systems.zkSNARK.grothBGM17.DistributedProver;
+import zk_proof_systems.zkSNARK.grothBGM17.ZKSnarkObjectReader;
+import zk_proof_systems.zkSNARK.grothBGM17.ZKSnarkObjectWriter;
+
+public class Prover {
+  public static void main(String[] args) throws IOException {
+    System.out.println("Distributed Prover (PoC implementation)");
+
+    var options = new Options();
+    options.addOption(new Option("h", "help", false, "Display this message"));
+    options.addOption(new Option("l", "local", false, "Run on local simulated cluster"));
+    options.addOption(new Option("t", "test", false, "Run trivial test to verify setup"));
+    options.addOption(new Option("p", "primary-size", false, "Size of primary input (1)"));
```

Comment:
Shouldn't the `hasArg` flag be set to `true` here?

Reply:
Yes it should.  Fixed.  Thanks.